pkg/controller/machine-set-boot-image/vsphere_helpers.go

```diff
@@ -0,0 +1,376 @@
+	file := item.Path
+
+	f, size, err := archive.Open(file)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+
+	opts := soap.Upload{
+		ContentLength: size,
+	}
+
+	return lease.Upload(ctx, item, f, opts)
+}
+
+func createNewVMTemplateWithNameForFailureDomain(ctx context.Context, providerSpec *machinev1beta1.VSphereMachineProviderSpec, failureDomain osconfigv1.VSpherePlatformFailureDomainSpec, finder *find.Finder, client *govmomi.Client, tagManager *tags.Manager, name, ovaPath, infraID string) error {
+	folder, err := finder.Folder(ctx, providerSpec.Workspace.Folder)
+	if err != nil {
+		return fmt.Errorf("failed to find folder: %w", err)
+	}
+
+	// OVF Descriptor
+	archive := &importer.TapeArchive{Path: ovaPath}
+	ovfDescriptor, err := importer.ReadOvf("*.ovf", archive)
+	if err != nil {
+		return debugCorruptOva(ovaPath, err)
+	}
+
+	ovfEnvelope, err := importer.ReadEnvelope(ovfDescriptor)
+	if err != nil {
+		return fmt.Errorf("failed to parse ovf: %w", err)
+	}
+
+	// The fcos ova enables secure boot by default, this causes
+	// scos to fail once
+	secureBoot := checkOvaSecureBoot(ovfEnvelope)
+
+	// The RHCOS OVA only has one network defined by default
+	// The OVF envelope defines this.  We need a 1:1 mapping
+	// between networks with the OVF and the host
+	if len(ovfEnvelope.Network.Networks) != 1 {
+		return fmt.Errorf("expected the OVA to only have a single network adapter")
+	}
+
+	cluster, err := finder.ClusterComputeResource(ctx, failureDomain.Topology.ComputeCluster)
+	if err != nil {
+		return fmt.Errorf("failed to find compute cluster: %w", err)
+	}
+
+	clusterHostSystems, err := cluster.Hosts(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to get cluster hosts: %w", err)
+	}
+
+	if len(clusterHostSystems) == 0 {
+		return fmt.Errorf("the vCenter cluster %s has no ESXi nodes", failureDomain.Topology.ComputeCluster)
+	}
+
+	resourcePool, err := finder.ResourcePool(ctx, providerSpec.Workspace.ResourcePool)
+	if err != nil {
+		return fmt.Errorf("failed to find resource pool: %w", err)
+	}
+
+	networkPath := path.Join(cluster.InventoryPath, failureDomain.Topology.Networks[0])
+
+	networkRef, err := finder.Network(ctx, networkPath)
+	if err != nil {
+		return fmt.Errorf("failed to find network: %w", err)
+	}
+	datastore, err := finder.Datastore(ctx, providerSpec.Workspace.Datastore)
+	if err != nil {
+		return fmt.Errorf("failed to find datastore: %w", err)
+	}
+
+	// Create mapping between OVF and the network object
+	// found by Name
+	networkMappings := []types.OvfNetworkMapping{{
+		Name:    ovfEnvelope.Network.Networks[0].Name,
+		Network: networkRef.Reference(),
+	}}
+
+	// This is a very minimal spec for importing an OVF.
+	cisp := types.OvfCreateImportSpecParams{
+		EntityName:       name,
+		NetworkMapping:   networkMappings,
+		DiskProvisioning: string(types.OvfCreateImportSpecParamsDiskProvisioningTypeThin),
```

Comment:
I am certain if we default this to thin a customer will consider this a bug

Reply:
ok resolved using looking at the installConfig